build/azure-pipelines/common/computeNodeModulesCacheKey.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+  *  Copyright (c) Microsoft Corporation. All rights reserved.
+  *  Licensed under the MIT License. See License.txt in the project root for license information.
+  *--------------------------------------------------------------------------------------------*/
+
+ 'use strict';
+
+ import * as fs from 'fs';
+ import * as path from 'path';
+ import * as crypto from 'crypto';
+
+ const ROOT = path.join(__dirname, '../../../');
+
+ function findFiles(location: string, pattern: string, result: string[]) {
+ 	const entries = fs.readdirSync(path.join(ROOT, location));
+
+ 	for (const entry of entries) {
+ 		const entryPath = `${location}/${entry}`;
+ 		let stat: fs.Stats;
+ 		try {
+ 			stat = fs.statSync(path.join(ROOT, entryPath));
+ 		} catch (err) {
+ 			continue;
+ 		}
+ 		if (stat.isDirectory()) {
+ 			findFiles(entryPath, pattern, result);
+ 		} else {
+ 			if (stat.isFile() && entry.endsWith(pattern)) {
+ 				result.push(path.join(ROOT, entryPath));
+ 			}
+ 		}
+ 	}
+ }
+
+ const shasum = crypto.createHash('sha1');
+
+ shasum.update(fs.readFileSync(path.join(ROOT, 'build/.cachesalt')));
+ shasum.update(fs.readFileSync(path.join(ROOT, '.yarnrc')));
+ shasum.update(fs.readFileSync(path.join(ROOT, 'remote/.yarnrc')));
+
+
+ const result: string[] = [];
+ findFiles('', 'yarn.lock', result);
+ result.forEach(f => shasum.update(fs.readFileSync(f)));
```

Comment:
They are doing the same thing if I am not wrong. Check this file out. https://github.com/microsoft/vscode/blob/28a246d9ec5d10d9eaca7c80e1c942b1768d696b/build/azure-pipelines/common/computeNodeModulesCacheKey.ts#L23

Reply:
They have a hardcoded array of all the dirs that may contain yarn.lock files. Instead, I am just scanning the entire project for them. The worst that can happen is we might have few more cache misses because of scanning additional yarn.lock files. This doesn't seem harmful to me. 

Check this file out https://github.com/microsoft/vscode/blob/28a246d9ec5d10d9eaca7c80e1c942b1768d696b/build/npm/dirs.js

Reply:
Hm, would love to know why they did this. Unfortunately the changes made to this engineering stuff seems to be very light on the explanations of why things are being changed.

I agree with your approach though, if we start running into issues with that though we can look into adopting what they did but I'm guessing it's more related to the different architectures and modes that they compile for and improving speeds than any actual issues with doing it this way.